client/blocks/signup-form/passwordless.jsx

```diff
@@ -0,0 +1,189 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import emailValidator from 'email-validator';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import LoggedOutForm from 'components/logged-out-form';
+import LoggedOutFormFooter from 'components/logged-out-form/footer';
+import ValidationFieldset from 'signup/validation-fieldset';
+import { onboardPasswordlessUser } from 'lib/signup/step-actions';
+import { recordTracksEvent } from 'state/analytics/actions';
+import Notice from 'components/notice';
+import { submitSignupStep } from 'state/signup/progress/actions';
+
+export class PasswordlessSignupForm extends Component {
+	static defaultProps = {
+		locale: 'en',
+	};
+
+	state = {
+		isSubmitting: false,
+		email: '',
+		errorMessages: null,
+	};
+
+	submitTracksEvent = ( isSuccessful, message ) =>
+		this.props.recordTracksEvent( 'calypso_signup_actions_onboarding_passwordless_login', {
+			is_successful: isSuccessful,
+			action_message: message,
+		} );
+
+	onFormSubmit = event => {
+		event.preventDefault();
+
+		if ( ! this.state.email || ! emailValidator.validate( this.state.email ) ) {
+			this.setState( {
+				errorMessages: [ this.props.translate( 'Please provide a valid email address.' ) ],
+				isSubmitting: false,
+			} );
+			this.submitTracksEvent( false, 'Please provide a valid email address.' );
+			return;
+		}
+
+		this.setState( {
+			isSubmitting: true,
+		} );
+
+		onboardPasswordlessUser( this.onboardPasswordlessUserCallback, {
+			email: typeof this.state.email === 'string' ? this.state.email.trim() : '',
+		} );
+	};
+
+	onboardPasswordlessUserCallback = ( error, response ) => {
+		if ( error ) {
+			const errorMessage = this.getErrorMessage( error );
+			this.setState( {
+				errorMessages: [ errorMessage ],
+				isSubmitting: false,
+			} );
+			this.submitTracksEvent( false, error.message );
+			return;
+		}
+
+		this.setState( {
+			errorMessages: null,
+			isSubmitting: false,
+		} );
+
+		this.submitStep( response );
+	};
+
+	getErrorMessage( errorObj = { error: null, message: null } ) {
+		const { translate } = this.props;
+
+		switch ( errorObj.error ) {
+			case 'already_taken':
+			case 'already_active':
+				return (
+					<>
+						{ translate( 'An account with this email address already exists.' ) }
+						&nbsp;
+						{ translate( 'If this is you {{a}}log in now{{/a}}.', {
```

Comment:
`If this is you...` feels odd to me. That said, I'm not certain which copy to suggest here instead.  Maybe something like:
`If you have access to this account, _log in now_.`
...but that doesn't exactly roll off the tongue, either.

Reply:
Another option would be to provide a button to kick off a magic login email from here.